File: Frontend/src/types/customer.ts
export enum CustomerType {
  PessoaFisica = 1,
  PessoaJuridica = 2,
}

export interface CustomerResponse {
  id: string;
  name: string;
  document: string;
  customerType: CustomerType;
  birthDate: string;
  phone: string;
  email: string;
  zipCode: string;
  street: string;
  number: string;
  neighborhood: string;
  city: string;
  state: string;
  stateRegistration: string | null;
  isStateRegistrationExempt: boolean;
  createdAt: string;
  updatedAt: string | null;
}

export interface CreateCustomerCommand {
  name: string;
  document: string;
  customerType: CustomerType;
  birthDate: string;
  phone: string;
  email: string;
  zipCode: string;
  street: string;
  number: string;
  neighborhood: string;
  city: string;
  state: string;
  stateRegistration: string | null;
  isStateRegistrationExempt: boolean;
}

export interface UpdateCustomerCommand {
  id: string;
  name: string;
  phone: string;
  email: string;
  zipCode: string;
  street: string;
  number: string;
  neighborhood: string;
  city: string;
  state: string;
  stateRegistration: string | null;
  isStateRegistrationExempt: boolean;
}

export interface CustomerEventResponse {
  id: string;
  customerId: string;
  eventType: string;
  payload: string;
  occurredAt: string;
  occurredBy: string;
}

File: Backend/CRM.Corporativo.Domain/Models/Customer.cs
using CRM.Corporativo.Domain.Enums;
using CRM.Corporativo.Domain.Models.Base;

namespace CRM.Corporativo.Domain.Models;

public class Customer : Entity
{
    public string Name { get; set; } = string.Empty;
    public string Document { get; set; } = string.Empty;
    public CustomerTypeEnum CustomerType { get; set; }
    public DateTime BirthDate { get; set; }
    public string Phone { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string ZipCode { get; set; } = string.Empty;
    public string Street { get; set; } = string.Empty;
    public string Number { get; set; } = string.Empty;
    public string Neighborhood { get; set; } = string.Empty;
    public string City { get; set; } = string.Empty;
    public string State { get; set; } = string.Empty;
    public string? StateRegistration { get; set; }
    public bool IsStateRegistrationExempt { get; set; }
}
